backend/api/pipeline.proto

```diff
@@ -347,4 +356,7 @@ message PipelineVersion {
 
   // Input. Optional. Description for the pipeline version.
   string description = 8;
+
+  // Output. The spec format for this pipeline.
+  PipelineSpecFormat spec_format = 9;
```

Comment:
Although not directly related to this change, but when we are retrieving Pipeline Runs, there are two similar fields `pipeline_manifest` and `workflow_manifest` in https://www.kubeflow.org/docs/components/pipelines/reference/api/kubeflow-pipeline-api-spec/#/definitions/apiPipelineRuntime. And one of them contains argo workflow. Because we need pipeline_spec when getting the Pipeline Run detail, which field should I look out for the IR? And should we consider adding PipelineSpecFormat there as well?

Reply:
cc @capri-xiyue will implement the run API.

Yes, `workflow_manifest` is what we currently use. For getting IR, I think it's better to take a different approach, what do you think about getting the IR from a pipeline version?
Design: runs have res refs to connect with pipeline versions, then UI gets corresponding pipeline version to get the pipeline spec.

Reply:
> what do you think about getting the IR from a pipeline version?
> Design: runs have res refs to connect with pipeline versions, then UI gets corresponding pipeline version to get the pipeline spec.

It will be awesome if we can use one network request to get IR, even if it is for `pipeline versioned run`. Because there is a field pipeline_spec in `apiRun`: https://www.kubeflow.org/docs/components/pipelines/reference/api/kubeflow-pipeline-api-spec/#/definitions/apiRun, we can utilize this field to return the IR from pipeline version.

> We need to think about one-off runs, we may store the pipeline spec in the run details object in this case, and I think it's reasonable to also add a pipeline spec format field here.

Yes I agree with this approach, and it also created the opportunity for `pipeline versioned run` to be consistent in returning PipelineSpec from backend.